File: PairWithSum.java
int countPairs(int arr[], int target) {
        // Complete the function
        int l = 0;
        int r = arr.length - 1;
        int cnt = 0;
        while(l < r){
            int sum = arr[l] + arr[r];
            if(sum == target){
                ++cnt;
                int nextLeft = l + 1;
                while(nextLeft < r && arr[nextLeft] == arr[nextLeft - 1]){
                    ++cnt;
                    ++nextLeft;
                }
                --r;
            }
            else if(sum < target){
                ++l;
            }
            else{
                --r;
            }
        }
        return cnt;
    }

File: PairWithSum.cpp
int countPairs(vector<int> &arr, int target) {
        // Complete the function
        int count = 0;
        int n = arr.size();
        
        int left = 0, right = n-1;
        
        while(left < right){
            int sum = arr[left] + arr[right];
            
            if(sum == target){
                count += 1;
                
                int nextLeft = left + 1;
                
                while(nextLeft < right && arr[nextLeft] == arr[nextLeft - 1]){
                    count += 1;
                    ++nextLeft;
                }
                
                right -= 1;
                
            }else if(sum < target){
                left += 1;
            }else{
                right -= 1;
            }
        }
        
        return count;
    }
